### PR TITLE
Project Exorcism Episode 1 : Turf Wars

### PR DESCRIPTION
**Pull request by alsoandanswer**
Majority of sprites by Atteria, others by Wezzy and modified Bay sprites.

Mainly ports over Bay's tile smoothing and updates our old code. Funnily enough, both coded by Lohikar.
changes:
  - imageadd: "Adds new sprites for turfs, by Atteria."
  - bugfix: "Fixes paint colors being ultra desaturated when painted on tiles."
  - rscadd: "You can now paint linoleum, freezer tiles, reinforced tiles and wooden tiles."
  - rscadd: "Tile paint decals are now modularised, meaning they adapt to the tile they are on."
  - rscadd: "Adds decal blending modes, allowing decals to adapt to the type of tile they're painted on."
  - rscadd: "Ports Bay/Eris carpet/grass smoothing and edges."
  - rscdel: "Removes depreciated tiles. Maps have been tweaked to reflect this."
  - rscadd: "Updates character preview background to match the new tiles."

Disclaimer for reviewers : Large number of lines are mostly search and replace in maps and removal of hardcoded tiles, which shouldn't be there anyway.

Now, there's still a bunch of odds and ends of mapping stuff, mainly paint decals. 
But that's for a part two, I don't want to bloat this PR even further.

Gallery :
dreamseeker_j0AcRNgmEY
dreamseeker_FJiKCLvtIY
dreamseeker_87X17cNab7

Don't mind me, just testing the paint gun.
dreamseeker_Y9b36nUFTS
dreamseeker_IJX3YR6ou3
dreamseeker_51Znq2bVv4

Edit :
  - rscadd: "Damage overlays for tiles have been improved."
  Damage to tiles should look better and more dynamic. When you throw a frag, it actually looks like you did something.

dreamseeker_aim89NJDvw

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!wip

**Comment by alsoandanswer** [comment]
!review

**Comment by Greenjoe12345** [comment]
Would this allow for the different types of wood flooring Bay has? (Different types of wooden planks)

**Comment by alsoandanswer** [comment]
> Would this allow for the different types of wood flooring Bay has? (Different types of wooden planks)

I think we already have them. They're just not implemented properly, nor mapped in.

File: code/game/objects/items/devices/floor_painter.dm [hunk 1]
```diff
@@ -104,8 +101,12 @@
 		to_chat(user, "<span class='warning'>\The [src] can only be used on station flooring.</span>")
 		return
 
-	if(!F.flooring || !F.flooring.can_paint || F.broken || F.burnt)
-		to_chat(user, "<span class='warning'>\The [src] cannot paint broken or missing tiles.</span>")
+	if(!F.flooring.can_paint)
+		to_chat(user, "<span class='warning'>\The [src] cannot paint this type of flooring.</span>")
```
Comment (Generalcamo) [inline code/game/objects/items/devices/floor_painter.dm:105]:
```suggestion
		to_chat(user, SPAN_WARNING("\The [src] cannot paint this type of flooring."))
```

File: code/game/objects/items/devices/floor_painter.dm [hunk 2]
```diff
@@ -104,8 +101,12 @@
...
+		return
+
+	if(!F.flooring || F.broken || F.burnt)
+		to_chat(user, "<span class='warning'>\The [src] cannot paint damaged or missing tiles.</span>")
```
Comment (Generalcamo) [inline code/game/objects/items/devices/floor_painter.dm:109]:
```suggestion
		to_chat(user, SPAN_WARNING("\The [src] cannot paint damaged or missing tiles."))
```

File: code/game/objects/structures/lattice.dm [hunk 3]
```diff
@@ -129,11 +129,35 @@
...
 	var/base_icon_state = "grate_dark"
 	var/damaged = FALSE
 
-/obj/structure/lattice/catwalk/indoor/grate/old
-	icon_state = "grate_dark_old"
+/obj/structure/lattice/catwalk/indoor/grate/attackby(obj/item/C, mob/user)
+	if(C.iswelder() && damaged)
+		var/obj/item/weldingtool/WT = C
+		if(C.use_tool(src, user, 5, volume = 50) && WT.use(1, user))
+			to_chat(user, SPAN_NOTICE("You slice apart the [src] leaving nothing useful behind."))
```
Comment (Generalcamo) [inline code/game/objects/structures/lattice.dm:140]:
```suggestion
			user.visible_message(
				SPAN_NOTICE("\The user slices apart \the [src], leaving nothing useful behind."),
				SPAN_NOTICE("You slice apart \the [src], leaving nothing useful behind."),
				SPAN_NOTICE("You hear the sound of a welder, slicing apart metal.")
			)
```

**Review by Generalcamo — CHANGES_REQUESTED** [review]

**Comment by remona-minett** [comment]
Personally I'm not a fan of the way the floor tiles look, they are a little visually busy. I wonder if a poll on before and after on some of the larger changes would be good.

**Comment by alsoandanswer** [comment]
https://forums.aurorastation.org/topic/18188-project-exorcism-episode-1-turf-wars-feedback-thread/

**Comment by NonQueueingMatt** [comment]
merged into other PR

**Pull request closed without merging**